dotnet/src/Connectors/Connectors.AI.Oobabooga/Completion/OobaboogaCompletionBase.cs

```diff
@@ -2,43 +2,30 @@
 
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using System.Net.Http;
 using System.Net.WebSockets;
-using System.Runtime.CompilerServices;
 using System.Text;
-using System.Text.Json;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
-using Microsoft.SemanticKernel.AI;
-using Microsoft.SemanticKernel.AI.TextCompletion;
 using Microsoft.SemanticKernel.Diagnostics;
 
-namespace Microsoft.SemanticKernel.Connectors.AI.Oobabooga.TextCompletion;
+namespace Microsoft.SemanticKernel.Connectors.AI.Oobabooga.Completion;
 
-/// <summary>
-/// Oobabooga text completion service API.
-/// Adapted from <see href="https://github.com/oobabooga/text-generation-webui/tree/main/api-examples"/>
-/// </summary>
-public sealed class OobaboogaTextCompletion : ITextCompletion
+public abstract class OobaboogaCompletionBase
 {
-    public const string BlockingUriPath = "/api/v1/generate";
-    private const string StreamingUriPath = "/api/v1/stream";
-
-    private readonly UriBuilder _blockingUri;
-    private readonly UriBuilder _streamingUri;
-    private readonly HttpClient _httpClient;
-    private readonly Func<ClientWebSocket> _webSocketFactory;
-    private readonly bool _useWebSocketsPooling;
-    private readonly int _maxNbConcurrentWebSockets;
-    private readonly SemaphoreSlim? _concurrentSemaphore;
-    private readonly ConcurrentBag<bool>? _activeConnections;
-    private readonly ConcurrentBag<ClientWebSocket> _webSocketPool = new();
-    private readonly int _keepAliveWebSocketsDuration;
-    private readonly ILogger? _logger;
+    public const string HttpUserAgent = "Microsoft-Semantic-Kernel";
```

Comment:
We use `Telemetry.HttpUserAgent` from `Microsoft.SemanticKernel.Diagnostics` to align user-agent value across all connectors.

Reply:
Done